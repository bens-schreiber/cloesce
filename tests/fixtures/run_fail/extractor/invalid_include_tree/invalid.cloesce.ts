// @ts-nocheck

@D1
export class Foo {
  @PrimaryKey
  id: number;

  @DataSource
  bar: number; // invalid data source type
}
