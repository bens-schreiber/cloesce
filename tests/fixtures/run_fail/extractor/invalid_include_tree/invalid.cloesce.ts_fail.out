==== CLOESCE ERROR ====
Error [InvalidIncludeTree]: Invalid Include Tree
Phase: TypeScript IDL Extraction
Context: Foo.bar 
@DataSource
  bar: number;

Suggested fix: Include trees must only contain references to a model's navigation properties.
